pkg/controller/storagecluster/reconcile.go

```diff
@@ -1277,3 +1301,36 @@ func newCleanupJob(sc *ocsv1.StorageCluster) *batchv1.Job {
 
 	return job
 }
+
+//convertLabelToNodeSelector returns NodeSelectorTerm type from a given LabelSelector
+func convertLabelToNodeSelector(labelSelector metav1.LabelSelector) corev1.NodeSelectorTerm {
+	term := corev1.NodeSelectorTerm{}
+	for key, value := range labelSelector.MatchLabels {
+		req := corev1.NodeSelectorRequirement{}
+		req.Key = key
+		req.Operator = corev1.NodeSelectorOpIn
+		req.Values = append(req.Values, value)
+		term.MatchExpressions = append(term.MatchExpressions, req)
+	}
+	numIter := len(labelSelector.MatchExpressions)
+	for i := 0; i < numIter; i++ {
+		req := corev1.NodeSelectorRequirement{}
+		req.Key = labelSelector.MatchExpressions[i].Key
+		req.Operator = corev1.NodeSelectorOperator(labelSelector.MatchExpressions[i].Operator)
+		req.Values = labelSelector.MatchExpressions[i].Values
+		term.MatchExpressions = append(term.MatchExpressions, req)
+	}
+	return term
+}
+
+// MatchingLabelsSelector filters the list/delete operation on the given label
+// selector (or index in the case of cached lists). A struct is used because
+// labels.Selector is an interface, which cannot be aliased.
+type MatchingLabelsSelector struct {
+	labels.Selector
+}
+
+// ApplyToList applies this configuration to the given list options.
+func (m MatchingLabelsSelector) ApplyToList(opts *client.ListOptions) {
```

Comment:
Add a comment here that this is to fulfill the interface. Maybe I'm just old, but I was confused for a while on this one. :stuck_out_tongue_winking_eye:

Reply:
Haha! Added!